contracts/rust/deployer/src/lib.rs

```diff
@@ -704,6 +826,44 @@ pub async fn is_contract(provider: impl Provider, address: Address) -> Result<bo
     Ok(true)
 }
 
+/// this depends on upgradeProxy.ts which has to be ran on a real network supported by the safeSDK
+pub async fn call_upgrade_proxy_script(
+    proxy_addr: String,
+    new_impl_addr: String,
+    init_data: String,
+    rpc_url: String,
+    safe_address: String,
+) -> Result<(String, bool), anyhow::Error> {
+    let script_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
```

Comment:
The `env!` directive is evaluated at compile time so this is likely not what we want here.